### PR TITLE
Permissions Fix: Removed owner actions from users with editor permissions

### LINKED ISSUE #258: Only allow notebook or group owners to change ownership, delete, toggle private/public and share notebooks

### PR DESCRIPTION
**Pull request by manfromjupyter**
Closes #258

### CODE REVIEW

**Comment by mcrutch** [comment]
Still poking at this but might need @jeffyjefflabs to chime in

**Comment by mcrutch** [comment]
Nevermind, found your problem, you aren't blocked anymore.

**Comment by manfromjupyter** [comment]
Oh hah, fail. Okay will finish it right now. Thank you!

File: app/models/user.rb [hunk 1]
```diff
@@ -262,6 +262,17 @@ def can_read?(notebook, use_admin=false)
       (use_admin && admin?)
   end
 
+  def owner(notebook)
```
Comment (mcrutch) [inline app/models/user.rb:265]: You wrote this function...but then don't use it in the verify_owner function?

Comment (manfromjupyter) [reply app/models/user.rb:265]: I used it for the view. I guess I forgot I might be able to use it for the controller. Will play with it in a few

Comment (manfromjupyter) [reply app/models/user.rb:265]: Got it now. Works perfectly. Good eye

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -546,6 +546,10 @@ def verify_edit_not_admin
     verify_edit(false)
   end
 
+  def verify_owner
+    raise User::Forbidden, 'Restricted to users with owner permissions.' unless @user.owner(@notebook)
```
Comment (mcrutch) [inline app/controllers/application_controller.rb:550]: Should also allow admin here

Comment (manfromjupyter) [reply app/controllers/application_controller.rb:550]: The method in the user model covers admin there already. 
Tested it and it checks out.

```def owner(notebook)
    if notebook != nil
      type = Notebook.find(notebook.id).owner_type
      if ((type == "User" && notebook.owner_id == id) || (type == "Group" && GroupMembership.where(user_id: id, group_id: Group.find(notebook.owner_id)).pluck(:owner)[0]) || admin?)
        return true
      end
    else
      return false
    end
  end```

**Pull request merged**